### PR TITLE
enable react

### PR DESCRIPTION
**Pull request by ochlodeep**

### CODE REVIEW

File: src/main/index.js [hunk 1]
```diff
@@ -15,4 +18,9 @@ console.log(tpl('daniel'));
 console.log(tpl2('daniel'));
 console.log(util);
 console.log(util.sayHello());
-vendorShimTest.doIt();
\ No newline at end of file
+vendorShimTest.doIt();
+
+ReactDOM.render(
+<HelloReact />,
```
Comment (epikhighs) [inline src/main/index.js:24]: how does this work w/o making this file a jsx?

File: webpack.config.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ const common = {
     module: {
         loaders: [
             {
-                test: /\.js$/, loader: 'babel-loader',
+                test: /\.jsx?$/, loader: 'babel-loader',
```
Comment (epikhighs) [inline webpack.config.js:39]: js & jsx files are needed by babel loader

File: webpack.config.js [hunk 3]
```diff
@@ -99,6 +99,7 @@ const common = {
         },
         modulesDirectories: ['node_modules', 'vendor_modules'],
         root: [p.kendo],
+        extensions: ['', '.js', '.jsx'], // allows importing these file types without extensions
```
Comment (epikhighs) [inline webpack.config.js:102]: this looks like it doesn't include the defaults....

https://webpack.github.io/docs/configuration.html#resolve-extensions

Comment (epikhighs) [inline webpack.config.js:102]: i wonder if it's better to not have extensions.  it's wierd that we don't add extensions for JS, but we do add them for other file types....  but for backwards compat i guess we should just leave .js alone

File: .babelrc [hunk 4]
```diff
@@ -6,6 +6,11 @@
     },
     "prod": {
       "plugins": []
+    },
+    "start-webpack-dev-server": {
```
Comment (epikhighs) [inline .babelrc:10]: so when we start webpack dev server from npm scripts, then it reads into this config and sets the react-hmre preset?

File: webpack.config.js (hunk 2 above)
Comment (ochlodeep) [reply webpack.config.js:39]: x? makes x optional so it includes both js and jsx

File: src/main/index.js (hunk 1 above)
Comment (ochlodeep) [reply src/main/index.js:24]: i don't know i just tried it and it worked

File: .babelrc (hunk 4 above)
Comment (ochlodeep) [reply .babelrc:10]: in theory

File: webpack.config.js (hunk 2 above)
Comment (epikhighs) [reply webpack.config.js:39]: smart!

**Pull request merged**